src/classes/player.js

```diff
@@ -67,6 +63,40 @@ class Player {
         break;
     }
   }
+
+  // Axis alligned collision detection
+  collide(object) {
+    const x1 = this.position.x;
+    const y1 = this.position.y;
+    const h1 = this.height;
+    const w1 = this.width;
+
+    const x2 = object.position.x;
+    const y2 = object.position.y;
+    const h2 = object.height;
+    const w2 = object.width;
+
+    return x1 <= x2 + w2
+      && x1 + w1 >= x2
+      && y1 <= y2 + h2
+      && h1 + y1 >= y2;
+  }
+
+  // Handle collision with other object
+  handleCollision(object) {
+    const y1 = this.position.y;
+    const h1 = this.height;
+
+    const y2 = object.position.y;
+
+    // The player is colliding to the object from above with a positive
+    // velocity. This means that we need to put the player on top of the object
+    // and set the `onGround` property to true.
+    if (y1 + h1 >= y2 && this.velocity.y >= 0) {
+      this.position.y = y2 - h1 + 1;
+      this.onGround = true;
```

Comment:
We could do this in a separate module, so that we make it more general. It could take two world objects as an argument, so the collision detection would look like this:

```
collide(object1, object2) {
    const x1 = object1.position.x;
    const y1 = object1.position.y;
    const h1 = object1.height;
    const w1 = object1.width;

    const x2 = object.position.x;
    const y2 = object.position.y;
    const h2 = object.height;
    const w2 = object.width;

    return x1 <= x2 + w2
      && x1 + w1 >= x2
      && y1 <= y2 + h2
      && h1 + y1 >= y2;
  }
```
For now, player would always be object1. We could then call the two colliding objects' handleCollision() function. At this point we should think about creating a "WorldObject" baseclass (since we already have step and render) or maybe an "ICollidable" interface (or perhaps both). This would be preferable since later we want to have objects that break and fall after collision.

Reply:
I did a major refactor in the repository. As part of this refactor, I addressed the changes you requested in this commit.